### ISSUE TITLE
Add an apt repository

### ISSUE THREAD

**Issue by Qik000** [Qalculate/libqalculate#192]
Not all Linux systems use Flatpak or Snap, there should be a ppa

**Comment by hanna-kn** [comment]
An AppImage is in progress (an initial attempt is available at https://send.firefox.com/download/4660920e72d8e5ed/#-EhR9mxhiXSp4LerbGbT7Q). 

PPA is, as I understand it, Ubuntu (and derivatives) specific and requires packages to be built for each version of Ubuntu. Similarly separate .deb packages would be needed for each supported version of Debian (and its derivatives). Users with Ubuntu 16.04, or later, can install the Snap package.

**Comment by Qik000** [comment]
> An AppImage is in progress (an initial attempt is available at https://send.firefox.com/download/4660920e72d8e5ed/#-EhR9mxhiXSp4LerbGbT7Q).
> 
> PPA is, as I understand it, Ubuntu (and derivatives) specific and requires packages to be built for each version of Ubuntu. Similarly separate .deb packages would be needed for each supported version of Debian (and its derivatives). Users with Ubuntu 16.04, or later, can install the Snap package.

You can just make a Debian repo and all users on debian-based distros including ubuntu will be able to use it. Also some people choose to not use snap and uninstall it because it clunky and launches slowly.

**Comment by hanna-kn** [comment]
I have now created a (partially) statically linked self-contained binary (linked against the latest libcurl, openssl, icu, and mpfr) which seem to run without on Ubuntu 14, 16, 18, and 19:
https://send.firefox.com/download/7d76bbdccb1e02ce/#vS6NB9jk3ajfNBL8oxEEMQ

**Comment by Qik000** [comment]
> I have now created a (partially) statically linked self-contained binary (linked against the latest libcurl, openssl, icu, and mpfr) which seem to run without on Ubuntu 14, 16, 18, and 19:
> https://send.firefox.com/download/7d76bbdccb1e02ce/#vS6NB9jk3ajfNBL8oxEEMQ

But its on FF Send, will the AppImage or better yet the DEB be available officially any time soon ?

**Comment by hanna-kn** [comment]
> But its on FF Send, will the AppImage or better yet the DEB be available officially any time soon ?

The self-contained binaries will be included in the next official release (on Monday).

**Comment by Qik000** [comment]
Self contained meaning AppImage ?

**Comment by hanna-kn** [comment]
> Self contained meaning AppImage ?

No. I discovered that I can create ordinary binaries, statistically link to libqalculate, libcurl, openssl, icu, and mpfr, which seem to run on most Linux distributions. They will be distributed as a compressed archive containing only the qalculate-gtk and qalc binaries, which after extraction can be run from anywhere and practically only require that gtk (>= 3.10) is installed. In most case this should give a better experience than an AppImage.

**Comment by Qik000** [comment]
> > Self contained meaning AppImage ?
> 
> No. I discovered that I can create ordinary binaries, statistically link to libqalculate, libcurl, openssl, icu, and mpfr, which seem to run on most Linux distributions. They will be distributed as a compressed archive containing only the qalculate-gtk and qalc binaries, which after extraction can be run from anywhere and practically only require that gtk (>= 3.10) is installed. In most case this should give a better experience than an AppImage.

Will a setup script be provided that creates a .desktop file like Tor Browser and Tutanota mail do ?
How will updates be handled ? Updating manually is a huge PITA

**Comment by hanna-kn** [comment]
> Will a setup script be provided that creates a .desktop file like Tor Browser and Tutanota mail do ?

No setup script is planned for the first version.

> How will updates be handled ? Updating manually is a huge PITA

This is unfortunately difficult to avoid with self-contained binaries (including AppImages). By default, the user will be notified when a new version is available.

**Comment by Qik000** [comment]
> This is unfortunately difficult to avoid with self-contained binaries (including AppImages). By default, the user will be notified when a new version is available.

And thats why the best option is to make a PPA

**Comment by hanna-kn** [comment]
~~No setup script is planned for the first version.~~

I will include a simple installation script (optionally copies the binaries, and installs a desktop file and an icon).

**Comment by hanna-kn** [comment]
The first official version is now available:
https://github.com/Qalculate/qalculate-gtk/releases/download/v3.10.0/qalculate-3.10.0-x86_64.tar.xz
_(Extract and run from anywhere. Includes both qalculate-gtk and qalc, and an optional installation script which installs a desktop file and an icon in ~/.local/share/.)_

I plan to automate the update process in the next version.

**Comment by dtantono** [comment]
it is been three years any updates?

**Comment by cD1rtX3** [comment]
I'm afraid it's been… [six] years.
